### PR TITLE
II-61: Call collections SDK 

### PR DESCRIPTION
**Pull request by aerim-yi**
[Jira Ticket](https://immutable.atlassian.net/browse/II-61?atlOrigin=eyJpIjoiYzM5NGJiYTI0NmY0NGJhMzkwZmIzMTQyZDMxZmNjZDgiLCJwIjoiaiJ9)

Unit testing will be added once the SDK is connected to the card component.

### CODE REVIEW

File: src/api/collections-api.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { getConfig, CollectionsApi, ListCollectionsResponse } from "@imtbl/core-sdk";
```
Comment (ABashkirtseva) [inline src/api/collections-api.tsx:1]: this file should be just `.ts` not `.tsx` as you don't have react components here.

File: src/api/collections-api.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+// Get a list of collections
+// Inspired from Immutable Core SDK documentation
+const GetYourCollections = async (): Promise<ListCollectionsResponse> => {
```
Comment (ABashkirtseva) [inline src/api/collections-api.tsx:5]: better to name GetCollections or listCollections, as they are not user's.

File: src/api/collections-api.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  const config = getConfig("ropsten");
+  const collectionsApi = new CollectionsApi(config.api);
+
+  const response = await collectionsApi.listCollections()
+
+  return response.data;
+};
+
+export default GetYourCollections
```
Comment (ABashkirtseva) [inline src/api/collections-api.tsx:14]: please remove default export and add export to your function as you name your file as collections-api, it means that you can have more than one method here.

File: src/pages/CollectionsPage.tsx [hunk 4]
```diff
@@ -1,16 +1,34 @@
-import { Fragment } from "react"
-import HeaderBar from "../components/HeaderBar/HeaderBar"
+import { Fragment, useEffect, useState } from "react"
 import { Container, Row, Col } from 'react-bootstrap';
+import { Collection } from "@imtbl/core-sdk";
+import HeaderBar from "../components/HeaderBar/HeaderBar"
+import GetYourCollections from "../api/collections-api";
 
 const CollectionsPage = () => {
+    const [result, setResult] = useState<[] | Array<Collection>>([])
```
Comment (ABashkirtseva) [inline src/pages/CollectionsPage.tsx:8]:
```suggestion
    const [result, setResult] = useState<Collection[]>([])
```

File: src/pages/CollectionsPage.tsx [hunk 5]
```diff
@@ -1,16 +1,34 @@
...
+
+    // inspired by this Stackflow answer:https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret
+    useEffect(() => {
+        async function fetchCollection() {
+            let response = await GetYourCollections()
+            setResult(response.result)
+        }
+        fetchCollection()
```
Comment (ABashkirtseva) [inline src/pages/CollectionsPage.tsx:16]:
```suggestion
    GetYourCollections().then((response) => {
      setResult(response.result)
    })
```

File: src/api/collections-api.tsx (hunk 2 above)
Comment (ABashkirtseva) [reply src/api/collections-api.tsx:5]: Please name function with first letter lower case, upper case uses only for classes

File: src/pages/CollectionsPage.tsx [hunk 6]
```diff
@@ -1,16 +1,34 @@
...
+
     return (
         <Fragment>
             <HeaderBar />
             <Container>
                 <Row>
-                    <Col>
-                        {/* This is where the cards will be displayed */}
-                    </Col>
+                    {result.length > 0 && result.map((item: Collection) => {
```
Comment (ABashkirtseva) [inline src/pages/CollectionsPage.tsx:24]: as your result is always array you don't need to check length > 0

**Comment by ABashkirtseva** [comment]
Please add tests

File: src/pages/CollectionsPage.tsx [hunk 7]
```diff
@@ -1,16 +1,34 @@
...
+    useEffect(() => {
+        async function fetchCollection() {
+            let response = await GetYourCollections()
+            setResult(response.result)
+        }
+        fetchCollection()
+    }, [])
+
     return (
         <Fragment>
```
Comment (pano-skylakis) [inline src/pages/CollectionsPage.tsx:20]: FYI there is a short-syntax for Fragments :)
https://reactjs.org/docs/fragments.html#short-syntax

File: src/pages/CollectionsPage.tsx (hunk 6 above)
Comment (pano-skylakis) [inline src/pages/CollectionsPage.tsx:24]: If `result.length` is NOT 0, then `result.length` will evaluate to TRUE.

We can shorten this to
```
{result.length && result.map((item: Collection) => {
```

https://www.freecodecamp.org/news/falsy-values-in-javascript/

File: src/pages/CollectionsPage.tsx [hunk 8]
```diff
@@ -1,16 +1,34 @@
...
-                        {/* This is where the cards will be displayed */}
-                    </Col>
+                    {result.length > 0 && result.map((item: Collection) => {
+                        return (
+                            <Col xs={12} md={4}>
+                                {/* We will add card component here */}
+                            </Col>
+                        )
+                    })}
+                    {result.length === 0 && <h1> loading.... </h1>}
```
Comment (pano-skylakis) [inline src/pages/CollectionsPage.tsx:31]: Similarly, if `result.length` IS 0, this will evaluate to FALSE, because 0 is a falsey value in JS.

We can shorten this to
```
{!result.length && <h1> loading.... </h1>}
```

**Review by pano-skylakis — APPROVED** [review]

File: src/pages/CollectionsPage.tsx (hunk 6 above)
Comment (aerim-yi) [reply src/pages/CollectionsPage.tsx:24]: @ABashkirtseva I am checking the length here as I want to display the loading message if the data is not there yet. See line 31. Once the data is there I show the cards.

**Review by nur221282 — APPROVED** [review]
